src/coreclr/vm/riscv64/stubs.cpp

```diff
@@ -1525,68 +1534,461 @@ static void LoadRegPair(BYTE* p, int reg1, int reg2, UINT32 offset)
+                    *(DWORD*)p = 0x01d50f33; p += 4;
+                    // ld  t5, 0(t5)
+                    *(DWORD*)p = 0x000f3f03; 
+
+                    // move to next indirection offset data
+                    dataOffset = dataOffset - 16 + 4; // subtract 16 as we have moved PC by 16 and add 4 as next data is at 4 bytes from previous data
+                }
+                else
+                {
+                    // ld  t5, #(pLookup->sizeOffset)(a0)
+                    *(DWORD*)p = 0x00053f03 | ((UINT32)pLookup->sizeOffset << 20); p += 4;
+                    dataOffset -= 4; // subtract 4 as we have moved PC by 4
+                }
+                // lui  t4, (slotOffset&0xfffff000)>>12
+                *(DWORD*)p = 0x00000eb7 | ((((UINT32)slotOffset & 0xfffff000) >> 12) << 12); p += 4;
+                // addi  t4, t4, (slotOffset&0xfff)
+                *(DWORD*)p = 0x000e8e93 | ((slotOffset & 0xfff) << 20); p += 4;
+                dataOffset -= 8;
+                // blt  t4, t5, CALL HELPER
+                pBLECall = p;       // Offset filled later
+                *(DWORD*)p = 0x01eec063; p += 4;
+                dataOffset -= 4;
+            }
+
+            if(pLookup->offsets[i] > 2047)
+            {
+                _ASSERTE(dataOffset < 2047);
+                // auipc  t4, 0
+                *(DWORD*)p = 0x00000297;
+                p += 4;
+                // lw  t4, dataOffset(t4)
+                *(DWORD*)p = 0x000eae83 | ((dataOffset & 0xfff) << 20);
+                p += 4;
+                // add a0, a0, t4
+                *(DWORD*)p = 0x01d50533; 
+                // lw  a0, 0(a0)
+                *(DWORD*)p = 0x00052503; 
+                p += 4;
+                // move to next indirection offset data
+                dataOffset = dataOffset - 16 + 4; // subtract 16 as we have moved PC by 16 and add 4 as next data is at 4 bytes from previous data
+            }
+            else
+            {
+                // offset must be 8 byte aligned
+                _ASSERTE((pLookup->offsets[i] & 0x7) == 0);
+                // ld  a0, #(pLookup->offsets[i])(a0)
+                *(DWORD*)p = 0x00053503 | ((UINT32)pLookup->offsets[i] << 20);
+                p += 4;
+                dataOffset -= 4; // subtract 4 as we have moved PC by 4
+            }
+        }
+
+        // No null test required
+        if (!pLookup->testForNull)
+        {
+            _ASSERTE(pLookup->sizeOffset == CORINFO_NO_SIZE_CHECK);
+            // ret
+            *(DWORD*)p = 0x00008067;
+            p += 4;
+        }
+        else
+        {
+            // beq  a0, x0, CALL HELPER:
+            *(DWORD*)p = 0x00050063;
+            p += 4;
+
+            // ret
+            *(DWORD*)p = 0x00008067;
+            p += 4;
+
+            // CALL HELPER:
+            if(pBLECall != NULL)
+                *(DWORD*)pBLECall |= ((UINT32)(p - pBLECall) << 8);
+
+            // ori  a0,t2,0
+            *(DWORD*)p = 0x0003e513;
```

Comment:
Nit: I think the proper instruction for moving registers on RISC-V is addi.

Reply:
It's more than just a nit — on an OoO CPU using the proper `addi rd,rs,0` instruction is virtually guaranteed to result in a register rename with no execution resources needed, while it would be likely many or most cores would not do the same for the `ori` version (or `xori ...,0` or `andi ...,-1` or `sll ...,0` etc).

Reply:
Done